relations/src/proof.rs

```diff
@@ -0,0 +1,102 @@
+use halo2_base::{
+    gates::{GateChip, GateInstructions},
+    poseidon::hasher::PoseidonHasher,
+    utils::BigPrimeField,
+    AssignedValue, Context,
+};
+
+use crate::poseidon_consts::{RATE, T};
+
+#[derive(Clone, Debug)]
+pub struct MerkleProof<F: BigPrimeField, const MAX_PATH_LEN: usize> {
+    pub path_shape: Vec<bool>,
```

Comment:
Why is this a `Vec` when we have the `const` generic?

Reply:
this is related to the other comment

Reply:
Done